### ISSUE TITLE
Tutorial examples can not be compiled, even though requirements are fulfilled

### ISSUE THREAD

**Issue by anates** [dealii/dealii#7946]
When trying to compile step-40 and step-55, I get the error message
```
  Error! This tutorial requires a deal.II library that was configured with
  the following options:

      DEAL_II_WITH_PETSC = ON
      DEAL_II_PETSC_WITH_COMPLEX = OFF
      DEAL_II_WITH_P4EST = ON

  or

      DEAL_II_WITH_TRILINOS = ON
      DEAL_II_WITH_P4EST = ON

  However, the deal.II library found at /opt/dealii_anates was configured
  with these options

      DEAL_II_WITH_PETSC = ON
      DEAL_II_PETSC_WITH_COMPLEX = TRUE
      DEAL_II_WITH_P4EST = ON
      DEAL_II_WITH_TRILINOS = ON

  which conflict with the requirements.

  One or both of the aforementioned combinations of prerequisites are not met
  by your installation, but at least one is required for this tutorial step.


-- Configuring incomplete, errors occurred!
```
As far as I can see, my installation fulfills the requirements for the second option, but cmake still fails. According to the error message, cmake should not fail, if I understand that correct?

**Comment by drwells** [comment]
Yup, this should work: both steps should default to using Trilinos in this case.

**Comment by bangerth** [comment]
Note that the violated condition is that if you use PETSc, then it needs to be configured with `DEAL_II_PETSC_WITH_COMPLEX = OFF`.

**Comment by tjhei** [comment]
#7956 should fix it for step-40.

**Comment by drwells** [comment]
@anates could you check to see if the solution in #7956 fixes the issue for you?

**Comment by anates** [comment]
@drwells Yes, it does. For step 55 I created https://github.com/dealii/dealii/pull/8014

**Comment by masterleinad** [comment]
Closed by #8014.
